packages/block-editor/src/components/rich-text/index.js

```diff
@@ -509,13 +525,27 @@ const RichTextContainer = compose( [
 			isSelected = selectionStart.clientId === clientId;
 		}
 
+		let extraProps = {};
+		if ( Platform.OS === 'native' ) {
+			// If the block of this RichText is unmodified then it's a candidate for replacing when adding a new block.
+			// In order to fix https://github.com/wordpress-mobile/gutenberg-mobile/issues/1126, let's blur on unmount in that case.
+			// This apparently assumes functionality the BlockHlder actually
+			const block = clientId && __unstableGetBlockWithoutInnerBlocks( clientId );
+			const shouldBlurOnUnmount = block && isSelected && isUnmodifiedDefaultBlock( block );
+			extraProps = {
+				blockIsSelected,
```

Comment:
Does `blockIsSelected` need to be set again? It was already set in the previous higher order component, no?

Reply:
@hypest do you know why we need to do it twice or this is a duplicate?

Reply:
Looks like superfluous. The second HOC is not modifying the value at all so, no need to extract and return it again.